From 0418e29d88110c9912de29e77634c5d1679f99fe Mon Sep 17 00:00:00 2001
From: Ivan Vecera <ivecera@redhat.com>
Date: Mon, 8 Sep 2014 11:40:42 -0400
Subject: [net] ethtool: Expand documentation of ethtool_ops::{get, set}_rxfh()

Message-id: <1410176446-10440-7-git-send-email-ivecera@redhat.com>
Patchwork-id: 91659
O-Subject: [RHEL7.1 PATCH 06/10] ethtool: Expand documentation of ethtool_ops::{get, set}_rxfh()
Bugzilla: 1138823
RH-Acked-by: Michal Schmidt <mschmidt@redhat.com>
RH-Acked-by: Jarod Wilson <jarod@redhat.com>
RH-Acked-by: Jiri Benc <jbenc@redhat.com>

BZ: #1138823

Upstream commit(s):
commit 678e30df2e5664619e06fcfea5490a476826d8fe
Author: Ben Hutchings <ben@decadent.org.uk>
Date:   Mon May 19 01:25:59 2014 +0100

    ethtool: Expand documentation of ethtool_ops::{get,set}_rxfh()

    Some corner-cases are not explained properly.

    Signed-off-by: Ben Hutchings <ben@decadent.org.uk>

Signed-off-by: Ivan Vecera <ivecera@redhat.com>
Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/include/linux/ethtool.h b/include/linux/ethtool.h
index 5b1a7f7..61c0a47 100644
--- a/include/linux/ethtool.h
+++ b/include/linux/ethtool.h
@@ -162,15 +162,16 @@ static inline u32 ethtool_rxfh_indir_default(u32 index, u32 n_rx_rings)
  * Will not be called if @get_rxfh_indir_size returns zero.
  * @get_rxfh: Get the contents of the RX flow hash indirection table and hash
  * key.
- * Will not be called if @get_rxfh_indir_size and @get_rxfh_key_size
- * returns zero.
+ * Will only be called if one or both of @get_rxfh_indir_size and
+ * @get_rxfh_key_size are implemented and return non-zero.
  * Returns a negative error code or zero.
  * @set_rxfh_indir: Set the contents of the RX flow hash indirection table.
  * Will not be called if @get_rxfh_indir_size returns zero.
- * @set_rxfh: Set the contents of the RX flow hash indirection table and
- * hash key.
- * Will not be called if @get_rxfh_indir_size and @get_rxfh_key_size
- * returns zero.
+ * @set_rxfh: Set the contents of the RX flow hash indirection table and/or
+ * hash key.  Either or both arguments may be %NULL if that attribute
+ * is not to be changed.
+ * Will only be called if one or both of @get_rxfh_indir_size and
+ * @get_rxfh_key_size are implemented and return non-zero.
  * Returns a negative error code or zero.
  * @get_channels: Get number of channels.
  * @set_channels: Set number of channels.  Returns a negative error code or
@@ -269,8 +270,8 @@ struct ethtool_ops {
  void   (*rh_reserved3)(void);
 #else
  u32 (*get_rxfh_key_size)(struct net_device *);
- int (*get_rxfh)(struct net_device *, u32 *, u8 *);
- int (*set_rxfh)(struct net_device *, u32 *, u8 *);
+ int (*get_rxfh)(struct net_device *, u32 *indir, u8 *key);
+ int (*set_rxfh)(struct net_device *, u32 *indir, u8 *key);
 #endif
  void   (*rh_reserved4)(void);
  void   (*rh_reserved5)(void);
-- 
1.7.1